packages/web-components/src/menu-item/menu-item.styles.ts

```diff
@@ -83,6 +83,96 @@ export const MenuItemStyles = css`
     :host(:active)::slotted(svg) {
         fill: ${neutralForegroundRestBehavior.var};
     }
+
+
+    :host([role="menuitemcheckbox"]),
+    :host([role="menuitemradio"]) {
+        display: grid;
+        grid-template-columns: auto auto 1fr minmax(42px, auto);
+        align-items: center;
+        min-height: 32px;
+    }
+
+    :host .input-container {
+        display: none;
+    }
+
+    :host([role="menuitemcheckbox"]) .input-container,
+    :host([role="menuitemradio"]) .input-container {
+        display: grid;
+        margin-inline-end: 10px;
+    }
+
+    :host([role="menuitemcheckbox"]) .start,
+    :host([role="menuitemradio"]) .start {
+        grid-column-start: 2;
+        margin-inline-end: 10px;
+    }
+
+    :host([role="menuitemcheckbox"]) .content,
+    :host([role="menuitemradio"]) .content {
+        grid-column-start: 3;
+    }
+
+    :host([role="menuitemcheckbox"]) .end,
+    :host([role="menuitemradio"]) .end {
+        grid-column-start: 4;
+    }
+
+    :host .checkbox,
+    :host .radio {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        position: relative;
+        width: 20px;
+        height: 20px;
+        box-sizing: border-box;
+        border: calc(var(--outline-width) * 1px) solid ${neutralForegroundRestBehavior.var};
+        outline: none;
+        margin-inline-start: 10px;
+    }
+
+    :host .checkbox {
+        border-radius: calc(var(--corner-radius) * 1px);
+    }
+
+    :host .radio {
+        border-radius: 999px;
+    }
+
+    :host .checkbox-indicator,
+    :host .radio-indicator,
+    ::slotted([slot="checkbox-indicator"]),
+    ::slotted([slot="radio-indicator"]) {
+        display: none;
+    }
+
+    :host([aria-checked="true"]) .checkbox-indicator,
+    :host([aria-checked="true"]) ::slotted([slot="checkbox-indicator"]) {
+        width: 100%;
+        height: 100%;
+        display: block;
+        fill: ${neutralForegroundRestBehavior.var};
+        pointer-events: none;
+    }
+
+    :host([aria-checked="true"]) .radio-indicator {
+        position: absolute;
+        top: 4px;
+        left: 4px;
+        right: 4px;
+        bottom: 4px;
+        border-radius: 999px;
+        display: block;
+        background: ${neutralForegroundRestBehavior.var};
+        pointer-events: none;
+    }
+
+    :host([aria-checked="true"]) ::slotted([slot="radio-indicator"]) {
+        display: block;
+        pointer-events: none;
+    }
 `.withBehaviors(
```

Comment:
to fix high contrast on the inputs you add these rule sets to `forcedColorsStylesheetBehavior`
```css
            :host .checkbox,
            :host .radio{
                border-color: ${SystemColors.ButtonText};
                background: ${SystemColors.HighlightText};
            }
            :host([checked="true"]) .checkbox,
            :host([checked="true"]) .radio {
                background: ${SystemColors.HighlightText};
                border-color: ${SystemColors.HighlightText};
            }
            :host(:hover) .checkbox,
            :host(:hover) .radio,
            :host(:${focusVisible}) .checkbox,
            :host(:${focusVisible}) .radio,
            :host([checked="true"]:hover) .checkbox,
            :host([checked="true"]:hover) .radio,
            :host([checked="true"]:${focusVisible}) .checkbox,
            :host([checked="true"]:${focusVisible}) .radio {
                border-color: ${SystemColors.HighlightText};
            }
            :host([aria-checked="true"]) {
                background: ${SystemColors.Highlight};
                color: ${SystemColors.HighlightText};
            }
            :host([aria-checked="true"]) .checkbox-indicator,
            :host([aria-checked="true"]) ::slotted([slot="checkbox-indicator"]),
            :host([aria-checked="true"]) ::slotted([slot="radio-indicator"]) {
                fill: ${SystemColors.Highlight};
            }
            :host([aria-checked="true"]) .radio-indicator {
                background: ${SystemColors.Highlight};
            }
            :host([checked="true"]) .start,
            :host([checked="true"]) .end {
                fill: ${SystemColors.HighlightText};
            }
```

Reply:
Thanks @khamudom - updated!